qiskit/circuit/quantumcircuit.py

```diff
@@ -1757,6 +1757,15 @@ def qasm(
             return None
         return out
 
+    def _repr_png_(self):
+        if self.width() > 50 or self.size() > 50 or self.depth() > 25:
+            return None
+        import io
+
+        img_byte = io.BytesIO()
+        self.draw("mpl").savefig(img_byte, format="PNG")
+        return img_byte.getvalue()
```

Comment:
I think we'll also need to catch the case that MPL isn't installed and return `None` in that case as well.  You can use the lazy checkers for that:

```python
from qiskit.utils import optionals

if not optionals.HAS_MATPLOTLIB:
    return None
with io.BytesIO() as img_byte:
    self.draw("mpl").savefig(img_byte, format="png")
    return img_byte.getvalue()
```